pkg/observation/observations.go

```diff
@@ -17,13 +19,40 @@ type Store interface {
 
 func NewRouter(s rest.Server, store Store) *chi.Mux {
 	r := chi.NewRouter()
+	r.Use(authorizationMiddleware(s,store))
 	r.Method("DELETE", "/{observationId}", deleteObservation(s, store))
 	// TODO: Use patch with upsert instead of PUT.
 	r.Method("PUT", "/{observationId}", updateObservation(s, store))
 	r.Method("GET", "/{observationId}", getObservation(s, store))
 	return r
 }
+func  authorizationMiddleware(s rest.Server,store Store) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return s.NewHandler(func(w http.ResponseWriter, r *http.Request) *rest.Error {
+			observationId := chi.URLParam(r, "observationId")
 
+			// Verify user access to the school
+			session, ok := auth.GetSessionFromCtx(r.Context())
+			if !ok {
+				return auth.NewGetSessionError()
+			}
+			observation, err := store.GetObservation(observationId)
+			if err == pg.ErrNoRows {
+				return &rest.Error{http.StatusNotFound, "We can't find the specified observation", err}
+			} else if err != nil {
+				return &rest.Error{http.StatusInternalServerError, "Failed getting observation", err}
+			}
+
+			// Check if user is related to the school
+			if session.UserId!=observation.CreatorId{
```

Comment:
We want to allow all users related to a school to be able to edit all observations that relate to the students enrolled in that school. I don't think checking through the `CreatorId` is enough. We would need to check, all the way up to which school the observation is related to and then verify if the user is a part of that school.

`CreatorId` contains the id of the original user that created the observation. If we were to check through the `CreatorId`, users wouldn't be able to edit/delete the observation created by other users, even though they're in the same school.